compute_sdk/tests/unit/conftest.py

```diff
@@ -35,3 +41,13 @@ def run_in_tmp_dir(tmp_path):
     os.chdir(tmp_path)
     yield tmp_path
     os.chdir(pwd)
+
+
+@pytest.fixture
+def mock_gc_home(mocker):
+    home = mocker.patch("globus_compute_sdk.sdk._environments._home")
+    home.return_value = pathlib.Path(MOCK_HOME_DIR)
+
+    yield ensure_compute_dir().absolute()
+
+    shutil.rmtree(MOCK_HOME_DIR)
```

Comment:
I think I tried using fs initially and it had issues, will try again (several refactors may have changed situation).

Reply:
I believe that for the tests involved fs is in play.  Regardless, especially since a couple of refactors have pulled the rug from under this code, consider `mock.patch.dict`:

```python
def mock_gc_home():
    env = os.environ.copy()
    env["GLOBUS_COMPUTE_USER_DIR"] = MOCK_HOME_DIR
    with mock.patch.dict(os.environ, env):
        yield ensure_compute_dir().absolute()
```

Reply:
There was a lot of redirecting output in the diagnostic and that produced strange issues in pytest, so I had to resort to 'real' local temporary files instead.

Reply:
I'm not aware of the specifics your referring to, but the existing diagnostic tests utilize `contextlib.redirect_stdout` to capture output.

E.g.,
```python
with contextlib.redirect_stdout(io.BytesIO()) as f:
    cat(str(bfile))()
```

Reply:
I'm not aware of the specifics you're referring to, but the existing diagnostic tests utilize `contextlib.redirect_stdout` to capture output.

E.g.,
```python
with contextlib.redirect_stdout(io.BytesIO()) as f:
    cat(str(bfile))()
```

Reply:
> I had to resort to 'real' local temporary files instead.

`tmp_path` creates a temporary directory on the filesystem and automatically cleans it up.

Reply:
> `tmp_path` creates a temporary directory on the filesystem and automatically cleans it up.

Ignoring the other parts of the conversation; being pedantic for a minute ... `tmp_path` provides a path to a test-specific directory that is created just for this test.  But I don't think it's cleaned-up at the end of _this_ test suite.  By default, pytest keeps the last N (3 I think?) runs, and cleans them in a rolling fashion.  For me, they're stored in `/tmp/pytest-of-kevin/...`.

Reply:
Changed to using tmp_path instead of the custom directory under tests/